src/utils/lollipop.ts

```diff
@@ -166,25 +218,41 @@ export const extractLollipopLocalsFromLollipopHeaders = (
                 ),
             eventLog.taskEither.errorLeft((errorResponse) => [
               `The lollipop function service returns an error | ${errorResponse.kind}`,
-              {
+              withoutUndefinedValues({
                 assertion_ref: assertionRef,
-                fiscal_code: sha256(fiscalCode),
+                fiscal_code: fiscalCode ? sha256(fiscalCode) : undefined,
                 name: LOLLIPOP_SIGN_ERROR_EVENT_NAME,
                 operation_id: operationId,
-              },
+              }),
             ])
           )
         )
       )
     ),
+    TE.chainFirst(({ assertionRef, operationId, lcParams }) =>
+      pipe(
+        O.fromNullable(fiscalCode),
+        O.map(() => TE.of(true)),
+        O.getOrElse(() =>
+          validateAssertionRefForUser(
+            sessionStorage,
+            lcParams.fiscal_code,
+            operationId,
+            assertionRef
+          )
+        )
+      )
+    ),
     TE.map(
-      (lcParams) =>
+      ({ lcParams }) =>
         ({
           ["x-pagopa-lollipop-assertion-ref"]: lcParams.assertion_ref,
           ["x-pagopa-lollipop-assertion-type"]: lcParams.assertion_type,
           ["x-pagopa-lollipop-auth-jwt"]: lcParams.lc_authentication_bearer,
           ["x-pagopa-lollipop-public-key"]: lcParams.pub_key,
-          ["x-pagopa-lollipop-user-id"]: fiscalCode,
+          // It's possible to improve security by verify that the fiscal code from
```

Comment:
```suggestion
          // It's possible to improve security by verifying that the fiscal code from
          // the authorization is equal to the one from the lollipop function
```

Reply:
Fixed in [06a2ed3](https://github.com/pagopa/io-backend/pull/1033/commits/06a2ed3c4f90691ff80f51bb2b091ad054e62149)